### DISCUSSION TITLE
Should I be using URL-and-SHA-256 style or tag-and-revision style when creating formulas?

### DISCUSSION THREAD

**Discussion by Southpaw1496** [Homebrew/discussions#3319] [Tap maintenance and Homebrew development] [answered]
I've been following the [Formula Cookbook](https://docs.brew.sh/Formula-Cookbook) to create a formula, and it says to directly link to a source tarball when using `brew create`. While making it, I was wondering what the process of updating it would be like, and in my research I found the [`bump-formula-pr`](https://docs.brew.sh/Manpage#bump-formula-pr-options-formula) command, which references "URL-and-SHA-256 style specification" and "tag-and-revision style specification". I also found that the [`create`](https://docs.brew.sh/Manpage#create-options-url) command has the `--HEAD` option to link the repository instead of the source tarball, which I assume is for tag-and-revision-style.

Which style is best to use, and if tag-and-revision is best, should the cookbook be updated?

**Comment by SMillerDev** [comment, marked as answer]
It's much more effort/data to check out a repo than it is to download a tarbal so if possible homebrew-core prefers tarbals.
